shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -24,17 +26,22 @@ async fn test_hotreload_basic_valkey_connection() {
 
 #[tokio::test]
 async fn test_dual_shotover_instances_with_valkey() {
+    let socket_path = "/tmp/test-hotreload.sock";
     let _compose = docker_compose("tests/test-configs/hotreload/docker-compose.yaml");
     let shotover_a = shotover_process("tests/test-configs/hotreload/topology.yaml")
         .with_hotreload(true)
+        .with_hotreload_socket_path(socket_path)
         .with_config("tests/test-configs/shotover-config/config_metrics_disabled.yaml")
         .start()
         .await;
     let client_a = Client::open("valkey://127.0.0.1:6380").unwrap();
     let mut con_a = client_a.get_connection().unwrap();
     let _: () = con_a.set("key_from_a", "value_from_a").unwrap();
+    sleep(Duration::from_secs(5)).await;
```

Comment:
This should not be needed.
If the test has a chance of failing we should alter the implementation to ensure that shotover waits for the hot reload init to complete before shotover declares itself to be ready.

Reply:
After looking at the code, to fix this issue you'll probably to setup a [tokio oneshot](https://docs.rs/tokio/latest/tokio/sync/oneshot/index.html). Pass the sender to the server task, and hold the receiver in main task, then you can signal that the server is setup in the server task and wait for a message to come through in the main task.